snps: nsim: arc_v: update RMX100 configuration and naming

AFOliveira opened the pull request:
This PR updates the Synopsys nSIM ARC-V RMX100 platform with improved naming conventions - that match arc classic - and updated hardware configuration and nsim options.

Changes include:
- Rename SOC configuration symbols to include vendor and platform information for better clarity and to prevent naming conflicts
- Update CLINT base address to match the default RMX100 configuration
- Update nSIM MMIO base address for proper memory mapping
- Remove PLIC support as ARC-V RMX only supports APLIC

These changes align the platform with the latest ARC-V RMX100 hardware specifications.

AFOliveira committed:
soc: snps: nsim: arc_v: rename SOC configuration symbols

Rename SOC configuration symbols from generic names to more specific
ones that include the vendor and platform information. This improves
clarity and prevents potential naming conflicts.

Changes:
- SOC_SERIES_RMX -> SOC_SERIES_NSIM_ARC_V_RMX
- SOC_RMX100 -> SOC_NSIM_ARC_V_RMX100

Signed-off-by: Afonso Oliveira <afonsoo@synopsys.com>

AFOliveira committed:
snps: nsim: arc_v: update RMX100 hardware configuration

Update the hardware configuration for the Synopsys nSIM ARC-V RMX100:

- Change CLINT base address from 0x2000000 to 0xa0010000. The CLINT
  is now created by default in the RMX100 configuration, with its
  MMIO base already configured to use this address.

- Update nSIM MMIO base address to 2560 to align with the new memory
  map requirements.

- Remove PLIC support as the ARC-V RMX architecture only supports
  APLIC (Advanced Platform-Level Interrupt Controller), not PLIC.
  Note that APLIC support is not yet available in Zephyr, so
  interrupt controller functionality will be limited until APLIC
  is implemented.

- Replace legacy MPU configuration with PMP (Physical Memory Protection):
  Remove mpu_version, mpu_regions, and mpu_granule properties and add
  ARCV_PMP interface with 16 regions and granule=1. The MPU options
  were legacy and PMP is the standard RISC-V memory protection mechanism.

These changes align with the latest ARC-V RMX100 hardware memory map
and architectural requirements.

Signed-off-by: Afonso Oliveira <afonsoo@synopsys.com>

AFOliveira commented on soc/snps/nsim/arc_v/rmx/CMakeLists.txt:
```diff
@@ -2,7 +2,7 @@
 if(COMPILER STREQUAL arcmwdt)
 # CCAC:
-  zephyr_compile_options_ifdef(CONFIG_SOC_SERIES_RMX
+  zephyr_compile_options_ifdef(CONFIG_SOC_SERIES_NSIM_ARC_V_RMX
```
Fixed

tagunil replied:
Nope, that's not what I meant. You see, this whole file is included only for the RMX series. It means you don't need additional checks to be sure it's RMX.

tagunil replied:
But if in the future we'll need to distinguish between, e.g. RMX100 and RMX500, then we could add some new checks.

AFOliveira replied:
I see what you mean about the file being RMX-specific already. But I'm wondering - doesn't it make sense to distinguish between base RMX extensions and optional ones? 

The way I split it, the unconditional options (Zicsr, Zifencei, Zihintpause, Zba, Zbb, Zbs, Zicbom) are the core extensions that any RMX variant should have, while the conditional ones (Zca, Zcb, Zcmp, Zcmt, Za, Zm) might be optional depending on the specific RMX implementation.

I know there's only one RMX in Zephyr right now, but wouldn't this approach help when/if we add more RMX variants later? It would make it clearer which extensions are baseline vs. variant-specific, right?

tagunil replied:
You have a point, but there is no sense in such separation until we introduce more RMX variants. And long-term I would prefer to move  RISC-V extension options out of here to the toolchain description files, as it's done for GCC.

AFOliveira replied:
Ok, now I believe it should be solved.

AFOliveira committed:
soc: snps: nsim: arc_v: rmx: remove redudant compiler filtering

Remove CONFIG_SOC_SERIES_NSIM_ARC_V_RMX conditional check from
zephyr_compile_options_ifdef and apply all extensions
unconditionally when using arcmwdt compiler.

Since this CMakeLists.txt is already SoC-specific, the
additional config check is redundant.

Signed-off-by: Afonso Oliveira <afonsoo@synopsys.com>

tagunil reviewed (approved):
Looks good to me.
